examples/fair_prep_results.py

```diff
@@ -0,0 +1,89 @@
+import os
+import sys
+import shutil
+import warnings
+warnings.simplefilter('ignore')
+
+from helper import extract_info
+
+sys.path.append(os.getcwd())
+
+from fp.traindata_samplers import CompleteData
+from fp.missingvalue_handlers import CompleteCaseAnalysis
+from fp.dataset_experiments import GermanCreditDatasetSexExperiment
+from fp.scalers import NamedStandardScaler, NamedMinMaxScaler
+from fp.learners import NonTunedLogisticRegression, LogisticRegression, DecisionTree, NonTunedLogisticRegression, NonTunedDecisionTree          
+from fp.post_processors import NoPostProcessing, RejectOptionPostProcessing, EqualOddsPostProcessing, CalibratedEqualOddsPostProcessing
+from fp.pre_processors import NoPreProcessing, DIRemover, Reweighing
+
+import numpy as np
+import matplotlib.pyplot as plt
+
+#creating list of parameters that we will alter to observe variations
+seeds = [0xbeef, 0xcafe, 0xdead, 0xdeadcafe, 0xdeadbeef, 0xbeefcafe, 0xcafebeef, 50, 583, 5278, 100000, 0xefac,0xfeeb, 0xdaed, 0xefacdaed, 0xfeebdead]
+learners = [NonTunedLogisticRegression(), LogisticRegression(), NonTunedDecisionTree(), DecisionTree()]
+processors = [(NoPreProcessing(), NoPostProcessing()), (DIRemover(1.0), NoPostProcessing()), (DIRemover(0.5), NoPostProcessing()), (Reweighing(), NoPostProcessing()),
+              (NoPreProcessing(), RejectOptionPostProcessing()), (NoPreProcessing(), CalibratedEqualOddsPostProcessing())]
+
+def calculate_metrics(seed, learner, pre_processor, post_processor):
+    '''
+    Experiment function to run the experiments
+    '''
+    exp = GermanCreditDatasetSexExperiment(
+        fixed_random_seed=seed,
+        train_data_sampler=CompleteData(),
+        missing_value_handler=CompleteCaseAnalysis(),
+        numeric_attribute_scaler=NamedStandardScaler(),
+        learners=[learner],
+        pre_processors=[pre_processor],
+        post_processors=[post_processor])
+    exp.run()
+
+def run_exp(seeds, learners, processors):
+    '''
+    This is the main driver function that calls the calculate_metrics to give metrices on combinations of various learners, pre and post processing techniques.
+    '''
+    accuracy, disp_imp, fnr, fpr = [], [], [], []
```

Comment:
why do we initialize these variables when we overwrite them in line 60?

Reply:
They are being initialized in the function to store values of various metrics. The same name has been used outside the function to store the same values. We can use the concept of global variables thus avoiding passing variables between functions, but read in a number of places that it is faster and better to pass lists than define variables globally.

Reply:
Ah, I thought they were overridden in the same function